src/cheri_sys_regs.sail

```diff
@@ -23,6 +25,8 @@ function legalize_ccsr(c : ccsr, v : xlenbits) -> ccsr = {
   let v = Mk_ccsr(v);
   let c = update_cap_idx(c, v.cap_idx());
   let c = update_cause(c, v.cause());
+  let c = update_gclgu(c, v.gclgu());
+  let c = update_gclgs(c, v.gclgs());
```

Comment:
Either the bits need to not exist for non-S-mode or we need a single register with per-mode views a la mstatus such that the bits are only accessible for >= S-mode. I'm a fan of the latter given it mirrors how a lot of the CSRs work; they tend to only have distinct registers when you can't cram all the per-mode bits into a single one. In that case the bit names should also have an `s` prefix to indicate the least-privileged mode that can see them, and I'd personally be inclined to move the `s` and `u` suffixes towards the front too to give `sugclg` and `ssgclg`, it feels more RISC-V-y to me, though there isn't really an equivalent to base it on.

Reply:
OK, but since we think that `xccsr.cap_cause` and `.cap_idx` are going away, can that happen before I refactor all this?